neo/utils.py

```diff
@@ -0,0 +1,856 @@
+
+        for epoch in epochs:
+            new_segments = cut_segment_by_epoch(
+                seg, epoch=epoch, reset_time=reset_time)
+            block.segments += new_segments
+
+        block.segments.remove(seg)
+    block.create_many_to_one_relationship(force=True)
+
+
+def cut_segment_by_epoch(seg, epoch, reset_time=False):
+    """
+    Cuts a Neo Segment according to a neo Epoch object
+
+    The function returns a list of neo Segments, where each segment corresponds
+    to an epoch in the neo Epoch object and contains the data of the original
+    Segment cut to that particular Epoch.
+
+    The resulting segments may either retain their original time stamps,
+    or can be shifted to a common time axis.
+
+    Parameters
+    ----------
+    seg: Neo Segment
+        The Segment containing the original uncut data.
+    epoch: Neo Epoch
+        For each epoch in this input, one segment is generated according to
+         the epoch time and duration.
+    reset_time: bool
+        If True the times stamps of all sliced objects are set to fall
+        in the range from 0 to the duration of the epoch duration.
+        If False, original time stamps are retained.
+        Default is False.
+
+    Returns:
+    --------
+    segments: list of Neo Segments
+        Per epoch in the input, a neo.Segment with AnalogSignal and/or
+        SpikeTrain Objects will be generated and returned. Each Segment will
+        receive the annotations of the corresponding epoch in the input.
+    """
+    if not isinstance(seg, neo.Segment):
+        raise TypeError(
+            'Seg needs to be of type neo.Segment, not %s' % type(seg))
+
+    if type(seg.parents[0]) != neo.Block:
+        raise ValueError(
+            'Segment has no block as parent. Can not cut segment.')
+
+    if not isinstance(epoch, neo.Epoch):
+        raise TypeError(
+            'Epoch needs to be of type neo.Epoch, not %s' % type(epoch))
+
+    segments = []
+    for ep_id in range(len(epoch)):
+        subseg = seg_time_slice(seg,
+                                epoch.times[ep_id],
+                                epoch.times[ep_id] + epoch.durations[ep_id],
+                                reset_time=reset_time)
+
+        # Add annotations of Epoch
+        for a in epoch.annotations:
+            if type(epoch.annotations[a]) is list \
+                    and len(epoch.annotations[a]) == len(epoch):
+                subseg.annotations[a] = copy.copy(epoch.annotations[a][ep_id])
+            else:
+                subseg.annotations[a] = copy.copy(epoch.annotations[a])
+
+        # Add array-annotations of Epoch
+        for key, val in epoch.array_annotations.items():
+            if len(val):
+                subseg.annotations[key] = copy.copy(val[ep_id])
+
+        segments.append(subseg)
+
+    return segments
+
+
+def seg_time_slice(seg, t_start=None, t_stop=None, reset_time=False, **kwargs):
```

Comment:
I think this would be more natural/consistent as a method of the `Segment` class (since `AnalogSignal` already has a `time_slice()` method).